File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day02IWasToldThereWouldBeNoMathTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, FunSuite, Matchers}

class Day02IWasToldThereWouldBeNoMathTest  extends FlatSpec with Matchers {
  "2015 Day 2 - Input" should "calculate answers" in {
    val dp = new Day02IWasToldThereWouldBeNoMath("/twenty_fifteen/Day02-IWasToldThereWouldBeNoMath-input.txt")
    dp.part1Answer should be(1606483)
    dp.part2Answer should be(3842356)
  }

  "2015 Day 2 - Example #1" should "calculate answers" in {
    val dp = new Day02IWasToldThereWouldBeNoMath("/twenty_fifteen/Day02-IWasToldThereWouldBeNoMath-example#1.txt")
    dp.part1Answer should be(101)
    dp.part2Answer should be(48)
  }
}

File: python-advent-of-code/tests/twenty_fifteen/day02_I_was_told_there_would_be_no_math_test.py
import unittest

from advent.twenty_fifteen.day02_I_was_told_there_would_be_no_math import Day02IWasToldThereWouldBeNoMath


class Day02IWasToldThereWouldBeNoMathTestCase(unittest.TestCase):
    def test_real(self):
        dp = Day02IWasToldThereWouldBeNoMath("data/twenty_fifteen/Day02-IWasToldThereWouldBeNoMath-input.txt")
        self.assertEqual(1606483, dp.part_1_answer)
        self.assertEqual(3842356, dp.part_2_answer)

    def test_example_1(self):
        dp = Day02IWasToldThereWouldBeNoMath("data/twenty_fifteen/Day02-IWasToldThereWouldBeNoMath-example#1.txt")
        self.assertEqual(101, dp.part_1_answer)
        self.assertEqual(48, dp.part_2_answer)


if __name__ == '__main__':
    unittest.main()
